src/backend/utils/adt/format_type.c

```diff
@@ -383,6 +383,22 @@ printTypmod(const char *typname, int32 typmod, Oid typmodout)
 
 		tmstr = DatumGetCString(OidFunctionCall1(typmodout,
 												 Int32GetDatum(typmod)));
+
+		/* Remove "without time zone" string from T-SQL's
+		 * sys.datetime2 and sys.smalldatetime types.
+		 *
+		 * This special handling can be removed once
+		 * 1. those types use their own typmod in/out functions, and
+		 * 2. there are no user objects using the old
+		 *    typmod in/out functions, i.e., in PG15.
+		 */
+		if (strcmp("sys.datetime2", typname) == 0 ||
+			strcmp("sys.smalldatetime", typname) == 0)
+		{
+			char *substr = strstr(tmstr, " without time zone");
+			if (substr)
+				*substr = '\0';
```

Comment:
are we sure that "without time zone" will always occur at the end of string? otherwise we can truncate real information here. May be just remove this substring to be on safe side

Reply:
I'm pretty sure that "without time zone" occurs at the end of the string for now, but let me follow your suggestion for more safety even in future changes.